src/components/Select/Select.tsx

```diff
@@ -0,0 +1,69 @@
+import { MenuToggle, SelectList, Select as SelectPF, SelectProps } from '@patternfly/react-core';
+import { ReactNode, useMemo } from 'react';
+
+import { TState } from 'hooks/useForm';
+
+export type TSelectOption = string | Object;
+
+interface ISelectProps<T extends TSelectOption> {
+  id?: SelectProps['id'];
+  isOpen: SelectProps['isOpen'];
+  onToggle: NonNullable<SelectProps['onOpenChange']>;
+  value?: T;
+  onChange: (event: Parameters<NonNullable<SelectProps['onSelect']>>[0], option: T) => void;
+  validated?: TState;
+  placeholder?: string;
+  onBlur?: SelectProps['onBlur'];
+  isToggleFullWidth?: boolean;
+  className?: string;
+  children: ReactNode | ReactNode[];
+}
+
+export const Select = <T extends TSelectOption>({
+  id,
+  className,
+  isOpen,
+  onToggle,
+  value,
+  onChange,
+  validated,
+  placeholder,
+  onBlur,
+  isToggleFullWidth = true,
```

Comment:
I understand there are 3 components we are using full width, but is there any other reason this default param needs to be altered to `true`? The default `isFullWidth` param of `MenuToggleProps` is `false`, and I can see you have to set `isToggleFullWidth=false` back in 2 components as well. I would suggest keeping the same default param logic with PF version components.

Reply:
The component will be used mostly as form select where the full width is needed.